packages/esm-billing-app/src/invoice/payments/initiate-payment/initiate-payment.component.tsx

```diff
@@ -14,45 +33,91 @@ export interface InitiatePaymentDialogProps {
 
 const InitiatePaymentDialog: React.FC<InitiatePaymentDialogProps> = ({ closeModal, bill }) => {
   const { t } = useTranslation();
+  const { mpesaCallbackUrl } = useConfig();
+  const { mflCodeValue } = useSystemSetting('facility.mflcode');
 
   const {
     control,
     handleSubmit,
     formState: { errors },
   } = useForm<any>({
     mode: 'all',
-    defaultValues: {},
+    defaultValues: {
+      billAmount: String(bill.totalAmount),
+    },
   });
 
-  const onSubmit = (data) => {
-    const payload = {
-      phoneNumber: data.phoneNumber,
-      amount: data.billAmount,
-      billUuid: bill.uuid,
-      referenceNumber: bill.receiptNumber,
-      callBackUrl: 'https://756e-105-163-1-73.ngrok-free.app/api/confirmation-url/',
-    };
-    initiateStkPush(payload).then(
-      (resp) => {
+  const onSubmit = async (data) => {
+    const validation = InitiatePaymentSchema.safeParse(data);
+    if (validation.success === false) {
+      const validationErrors = validation.error?.errors.map((error) => error.message);
+
+      validationErrors.forEach((error) => {
+        showSnackbar({
+          title: t('InitiatePaymentError', 'Initiate Payment Error'),
+          subtitle: error,
+          kind: 'error',
+          timeoutInMs: 3500,
+          isLowContrast: true,
+        });
+      });
+    } else {
+      try {
+        const shortCode = '174379';
+        const timeStamp = new Date()
+          .toISOString()
+          .replace(/[^0-9]/g, '')
+          .slice(0, -3);
+        const phoneNumber = formatPhoneNumber(data.phoneNumber);
+        const amountBilled = data.billAmount;
+        const passKey = 'bfb279f9aa9bdbcf158e97dd71a467cd2e0c893059b10f78e6b72ada1ed2c919';
+        const password = shortCode + passKey + timeStamp;
+        const callBackUrl = mpesaCallbackUrl;
+        const Password = Buffer.from(password).toString('base64');
+        const accountReference = `${mflCodeValue}#${bill.receiptNumber}`;
+
+        const payload = {
+          BusinessShortCode: shortCode,
+          Password: Password,
+          Timestamp: timeStamp,
+          TransactionType: 'CustomerPayBillOnline',
+          PartyA: phoneNumber,
+          PartyB: shortCode,
+          PhoneNumber: phoneNumber,
+          CallBackURL: callBackUrl,
+          AccountReference: accountReference,
+          TransactionDesc: 'HelloTest',
```

Comment:
```suggestion
          TransactionDesc: 'HelloTest',
```
I believe left by mistake